### PR TITLE
Adds TACOS page to docs and links to it from all mentionds of TACOS

### PR DESCRIPTION
**Pull request by Andrew-Hayes**
Resolves  https://github.com/opentofu/opentofu/issues/1666

## Target Release

1.9.0

## Checklist

- [x] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
- [x] I have not used an AI coding assistant to create this PR.
- [x] I have written all code in this PR myself OR I have marked all code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from.

### Go checklist

 

- [ ] I have run golangci-lint on my change and receive no errors relevant to my code.
- [ ] I have run existing tests to ensure my code doesn't break anything.
- [ ] I have added tests for all relevant use cases of my code, and those tests are passing.
- [ ] I have only exported functions, variables and structs that should be used from other packages.
- [ ] I have added meaningful comments to all exported functions, variables, and structs.

### Website/documentation checklist

- [x] I have locally started the website as [described here](https://github.com/opentofu/opentofu/blob/main/website/README.md) and checked my changes.

### CODE REVIEW

File: website/docs/intro/tacos.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+sidebar_position: 7
+sidebar_label: What are TACOS?
+description: |-
+  Learn all about TACOS.
+---
+
+# What are TACOS?
+
+'TACOS' stands for: TF Automation and Collaboration Software. They are platforms that perform, as part of their offering, OpenTofu runs to provision infrastructure, offering a collaboration-focused environment that makes it easier for teams to use OpenTofu together.
```
Comment (cam72cam) [inline website/docs/intro/tacos.mdx:10]:
```suggestion
TF Automation and Collaboration Software (TACOS) are platforms which allow teams to manage and orchestrate OpenTofu execution. They offer a wide variety of services to provide a streamlined and collaborative experience.
```

File: website/docs/intro/tacos.mdx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+# What are some examples?
+
+There are a few platforms that perform OpenTofu runs as part of their offering. Many of which are also our [supporters](../../../supporters).
```
Comment (cam72cam) [inline website/docs/intro/tacos.mdx:14]: This section is a bit harder to define.

```suggestion
There are a variety of platforms which offer OpenTofu support, both Open Source and Commercial. Many of these organizations support OpenTofu directly and can be found on our [supporters](../../../supporters) page.
```

**Comment by Andrew-Hayes** [comment]
Created issue https://github.com/opentofu/opentofu/issues/1842 to track adding further details to the page

**Review by cam72cam — APPROVED** [review]

**Review by Evi1Pumpkin — APPROVED** [review]
LGTM!

**Pull request merged**